agent/cache/cache.go

```diff
@@ -554,9 +554,15 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 		return ch
 	}
 
-	// If we already have an entry and it is actively fetching, then return
-	// the currently active waiter.
+	// If we already have an entry, it is actively fetching, and we don't need
+	// to revalidate the cache, then return the currently active waiter.
 	if ok && entry.Fetching {
+		// If we need to revalidate, return immediately
```

Comment:
```suggestion
		// If we need to revalidate, return with a closed channel
		// so the caller will immediately retry the cache
```

Reply:
Not sure if `fetch` has more than one caller but felt like the closed channel required some extra justification